skiko/src/awtMain/kotlin/org/jetbrains/skiko/swing/SkiaSwingLayer.kt

```diff
@@ -0,0 +1,123 @@
+package org.jetbrains.skiko.swing
+
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skiko.*
+import org.jetbrains.skiko.redrawer.RedrawerManager
+import java.awt.Graphics2D
+import java.awt.GraphicsConfiguration
+import javax.accessibility.Accessible
+import javax.swing.SwingUtilities.isEventDispatchThread
+
+/**
+ * Swing component that draws content provided by [skikoView] with GPU acceleration using Skia engine.
+ *
+ * Drawn content can be clipped by providing [ClipRectangle] to [clipComponents].
+ *
+ * This component can be used for better interop with Swing,
+ * so all Swing functionality like z-ordering, double-buffering etc. will be taken into account during rendering.
+ *
+ * But if no interop with Swing is needed, it is better to use [SkiaLayer] instead.
+ */
+@Suppress("unused") // used in Compose Multiplatform
+@ExperimentalSkikoApi
+open class SkiaSwingLayer(
+    skikoView: SkikoView,
+    analytics: SkiaLayerAnalytics = SkiaLayerAnalytics.Empty,
+) : SkiaSwingLayerComponent() {
+    internal companion object {
+        init {
+            Library.load()
```

Comment:
I do not see any `external` functions in this class. What is it for?

Reply:
> This is the same like 


It's required ONLY in wrapper classes that have native calls

Reply:
Where did you get this information? I see that no `ContextHandler`s have this call, it is only inside SkiaLayer

Reply:
I mean, for sure, I can put it in every file with an external call, but I can put it once where it is most probably will be needed.
I guess folk that wrote SkiaLayer also thought this way

Reply:
https://github.com/JetBrains/skiko/commit/55d72a8fdceea51a951f6cdc58b2c8b9c73f48e7

Reply:
Checking/loading native library makes sense only if you need to call native method from it. It's placed in `companion object/init` block for making sure that native calls might be made from here (that class)

Reply:
Okay, placing it in "umbrella" class works too, I was not sure that it's that case

Reply:
It is the case because we call skia and even some metal (will call directx, opengl) in nested classes